contracts/libraries/LibTypes.sol

```diff
@@ -2,119 +2,38 @@
-    // unlike the regular archaeologists, the sarcophagus already stores the
-    // single arweave archaeologist's address so there is no need to pass in the
-    // address to the finalizeSarcophagus function.
-    struct Signature {
-        uint8 v;
-        bytes32 r;
-        bytes32 s;
-    }
-
-    // Signature struct created to make passing in the signature argmuments into
-    // finalizedSarcophagus easier and to avoid the "stack too deep" error.
-    // Also attaching arachaeologist addresses so we can tie the signature back
-    // to the address in finalizeSarcophagus.
-    struct SignatureWithAccount {
-        address account;
-        uint8 v;
-        bytes32 r;
-        bytes32 s;
-    }
-
-    // SelectedArchaeologistData is the struct that is passed into the
-    // initializeSarcophagus function. Even though we don't need each storage
-    // fee of the archaeologist, the storage fee is included in the struct to
-    // reduce the stack size within the function, preventing the "stack too
-    // deep" error.
-    struct SelectedArchaeologistData {
-        address archAddress;
-        uint256 diggingFee;
-        bytes32 unencryptedShardDoubleHash;
-        uint8 v;
-        bytes32 r;
-        bytes32 s;
+    struct Sarcophagus {
+        // never zero - use for existence checks
+        uint256 resurrectionTime;
+        // todo: run gas cost evaluation on storing isCompromised vs looping through stored archaeologists and checking isAccused
+        bool isCompromised;
+        string name;
+        uint8 threshold;
+        uint256 maximumRewrapInterval;
+        string[2] arweaveTxIds;
+        address embalmerAddress;
+        address recipientAddress;
+        address[] cursedArchaeologistAddresses;
+        mapping(address => CursedArchaeologist) cursedArchaeologists;
     }
 
-    // ArchaeologistStorage is the struct that is stored in AppStorage under the
-    // sarcophagusArchaeologists mapping.
-    //
-    // The archaeologist address is left out since each archaeologist's address
-    // is stored on the sarcophagus object as an array.
-    //
-    // The storage fee is left out because we only need to store the storage fee
-    // of the archaeologist uploading to arweave, which will be stored directly
-    // on the sarcophagus.
-    struct ArchaeologistStorage {
+    struct CursedArchaeologist {
+        // never zero - use for existence checks
+        bytes32 doubleHashedKeyShare; // todo: we shouldn't need this and the doublehash->arch mapping
         bool isAccused;
         uint256 diggingFee;
-        uint256 diggingFeesPaid;
-        bytes32 unencryptedShardDoubleHash;
-        bytes unencryptedShard;
+        bytes rawKeyShare;
     }
 
-    // ArchaeologistProfile is used to store archaeologist profile data
     struct ArchaeologistProfile {
-        bool exists;
+        bool exists; // todo: use peerid.length instead of exists
```

Comment:
We may be able to get rid of this field and check for an empty peerId instead

Reply:
Agreed